src/main/java/com/skynest/models/EditBucketRequest.java

```diff
@@ -1,25 +1,33 @@
 package com.skynest.models;
 
 import com.skynest.utils.RandomGenerator;
-import lombok.AllArgsConstructor;
-import lombok.Builder;
-import lombok.Data;
-import lombok.NoArgsConstructor;
+import lombok.*;
 
 @Data
 @AllArgsConstructor
 @NoArgsConstructor
 @Builder
+@With
 public class EditBucketRequest {
     private String name;
     private String description;
     private Boolean isPublic;
 
-    public static EditBucketRequest generateValidEditBucketRequest() {
+    public static EditBucketRequest generateValidBucketEditRequest() {
```

Comment:
instead of having these 2 factory methods, the only difference between them being the value of a single field, consider having a single factory method that accepts an argument to set that field. This has the added benefit of having less code (duplication) with no apparent downsides